### ISSUE TITLE
PROPAGATORS=b3,tracecontext has confusing behavior

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-auto-instr-java#425]
If you configure:

`PROPAGATORS=b3,tracecontext`

then `b3` header extraction does not work, because `tracecontext` header extraction occurs afterwards and will overwrite any context extracted from `b3` headers with an invalid context.

From the javadoc for `DefaultContextPropagators.Builder.addHttpTextFormat()`:

```
One propagator per concern (traces, correlations, etc) should be added
```

I suggest that we log a warning if user supplies two propagators for the same concern (e.g. both "b3" and "tracecontext") and only apply the first one.

Reported on gitter: https://gitter.im/open-telemetry/opentelemetry-auto-instr-java?at=5ec66b83549761730b4b882d

See also https://github.com/open-telemetry/opentelemetry-specification/issues/496

**Comment by trask** [comment]
@carlosalberto I want to check that this proposal make sense to you before I tag it `help wanted`

**Comment by carlosalberto** [comment]
> I suggest that we log a warning if user supplies two propagators for the same concern (e.g. both "b3" and "tracecontext") and only apply the first one.

This sounds like a reasonable option for now.

(That being said, I have heard users wanting a `Propagator` that could supports many formats, i.e. `B3Propagator` **PLUS** `TraceContext` **PLUS** `Jaeger` etc, and for this case cooperation between them so they don't overwrite their values is needed - something I'm prototyping locally, but eventually, medium term, I hope we can support that)

**Comment by malafeev** [comment]
@trask could you assign it to me?

**Comment by malafeev** [comment]
fixed via #451, can be closed now.

**Comment by trask** [comment]
👍
